sway/desktop/layer_shell.c

```diff
@@ -212,6 +214,9 @@ void arrange_layers(struct sway_output *output) {
 	wl_list_for_each(seat, &input_manager->seats, link) {
 		seat_set_focus_layer(seat, topmost ? topmost->layer_surface : NULL);
 	}
+
+	arrange_windows(output->swayc);
+	transaction_commit_dirty();
```

Comment:
Calling `transaction_commit_dirty` here could lead to problems if there is other code that calls `arrange_layers` at a time when the tree is in an inconsistent state. I try to only call it directly in the handler functions to avoid this.

Also, there is other code in sway that calls `arrange_layers` followed by `arrange_windows`, so this would end up calling `arrange_windows` twice which isn't very efficient.

To be safe, I'd pull these out and call them from the relevant `handle_*` functions in this file.

BTW, I'd like to wire `transaction_commit_dirty` up to the event loop so it's run automatically, but I haven't looked into that yet.